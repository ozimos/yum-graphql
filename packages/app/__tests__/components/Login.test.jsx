/* global React:false, shallow:false toJson:false */

import { Login, mapStateToProps }
  from '../../src/components/login/ConnectedLogin';

const props = {
  dispatch: jest.fn(),
  authenticated: false,
};
describe('Login Component', () => {
  const setup = () => shallow(<Login {...props} />);

  it('should render without throwing an error', () => {
    const wrapper = setup();

    expect((wrapper)
      .exists(<form className="form" />)).toBe(true);
  });

  it('renders correctly', () => {
    const propsWithAuthentication = {
      dispatch: jest.fn(),
      authenticated: true
    };
    const propsWrongPassword = {
      dispatch: jest.fn(),
      loginError: { password: 'wrong password' }
    };
    let wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper = shallow(<Login {...propsWithAuthentication} />);
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper = shallow(<Login {...propsWrongPassword} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call `componentDidUpdate` on login error', () => {
    const serverFeedbackSpy = jest.spyOn(
      Login.prototype,
      'serverFeedback'
    );
    const componentDidUpdateSpy = jest.spyOn(
      Login.prototype,
      'componentDidUpdate'
    );
    serverFeedbackSpy.mockImplementation(() => jest.fn());
    const wrapper = setup();
    wrapper.setState({ prevSignupError: { password: 'wrong password' },
      newError: true });
    wrapper.instance().componentDidUpdate();
    expect(componentDidUpdateSpy).toHaveBeenCalled();
  });

  it('should call `handleLoginSubmit` on login form submit', () => {
    const wrapper = setup();

    const handleLoginSubmitSpy = jest.spyOn(
      wrapper.instance(),
      'handleLoginSubmit'
    );
    wrapper.instance().handleLoginSubmit();
    expect(handleLoginSubmitSpy).toHaveBeenCalled();
  });

  it('should call `disableLoginButton` on incorrect login form input ', () => {
    const wrapper = setup();

    const disableLoginButtonSpy = jest.spyOn(
      wrapper.instance(),
      'disableLoginButton'
    );
    wrapper.instance().disableLoginButton();
    expect(disableLoginButtonSpy).toHaveBeenCalled();
  });

  it('should call `enableLoginButton` on correct login form input', () => {
    const wrapper = setup();

    const enableLoginButtonSpy = jest.spyOn(
      wrapper.instance(),
      'enableLoginButton'
    );
    wrapper.instance().enableLoginButton();
    expect(enableLoginButtonSpy).toHaveBeenCalled();
  });

  it('should map state to props', () => {
    const initialState = {
      loginReducer: {
        authenticated: false,
        loginError: ''
      }
    };

    const tree = mapStateToProps(initialState);
    expect(tree).toMatchSnapshot();
  });
});

