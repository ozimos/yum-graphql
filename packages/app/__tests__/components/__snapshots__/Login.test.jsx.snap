// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders correctly 1`] = `
<div
  className="canvas"
>
  <div
    className="container2"
  >
    <header
      className="header"
    >
      <h2
        className="heading"
      >
        Book A Meal
      </h2>
    </header>
    <main>
      <div
        className="welcome"
      >
        <h4
          className="welcome-text"
        >
          Welcome!
        </h4>
        <h5>
          Sign in by entering the information below
        </h5>
      </div>
      <div
        className="form-box"
      >
        <Formsy
          className="form"
          disabled={false}
          getErrorMessage={[Function]}
          getErrorMessages={[Function]}
          getValue={[Function]}
          hasValue={[Function]}
          isFormDisabled={[Function]}
          isFormSubmitted={[Function]}
          isPristine={[Function]}
          isRequired={[Function]}
          isValid={[Function]}
          isValidValue={[Function]}
          mapping={null}
          onChange={[Function]}
          onError={[Function]}
          onInvalid={[Function]}
          onInvalidSubmit={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
          onValid={[Function]}
          onValidSubmit={[Function]}
          preventExternalInvalidation={false}
          resetValue={[Function]}
          setValidations={[Function]}
          setValue={[Function]}
          showError={[Function]}
          showRequired={[Function]}
          validationErrors={null}
        >
          <Formsy(Input)
            innerRef={[Function]}
            name="email"
            placeholder="Email"
            required={false}
            typeOfInput="email"
            validationError="This is not a valid email"
            validationErrors={
              Object {
                "isEmail": "This is not a valid email",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "input must be longer than 5 character",
              }
            }
            validations={
              Object {
                "isEmail": true,
                "maxLength": 48,
                "minLength": 5,
              }
            }
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="password"
            placeholder="Password"
            required={true}
            typeOfInput="password"
            validationError=""
            validationErrors={
              Object {
                "hasLowerCase": "Must have lowercase letter or you can
                     use a passphrase of minimum length 16 characters",
                "hasNumber": "Must have a number or you can
                     use a passphrase of minimum length 16 characters",
                "hasSpecialCharacter": "Must have a special character or 
                    you can use a passphrase of minimum length 16 characters",
                "hasUpperCase": "Must have uppercase letter or you can 
                    use a passphrase of minimum length 16 characters",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "Must have at least 8 characters",
              }
            }
            validations={
              Object {
                "hasLowerCase": [Function],
                "hasNumber": [Function],
                "hasSpecialCharacter": [Function],
                "hasUpperCase": [Function],
                "maxLength": 48,
                "minLength": 8,
              }
            }
          />
          <button
            className="btn login btn-disabled"
            disabled={true}
            type="submit"
          >
            Continue
          </button>
        </Formsy>
        <div
          className="stacked-text"
        >
          <Link
            replace={false}
            to="/signup"
          >
            <p
              id="signup"
            >
              Don't have an account? Click here to create one
            </p>
          </Link>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Login Component renders correctly 2`] = `
<div
  className="canvas"
>
  <div
    className="container2"
  >
    <header
      className="header"
    >
      <h2
        className="heading"
      >
        Book A Meal
      </h2>
    </header>
    <main>
      <div
        className="welcome"
      >
        <h4
          className="welcome-text"
        >
          Welcome!
        </h4>
        <h5>
          Sign in by entering the information below
        </h5>
      </div>
      <div
        className="form-box"
      >
        <Formsy
          className="form"
          disabled={false}
          getErrorMessage={[Function]}
          getErrorMessages={[Function]}
          getValue={[Function]}
          hasValue={[Function]}
          isFormDisabled={[Function]}
          isFormSubmitted={[Function]}
          isPristine={[Function]}
          isRequired={[Function]}
          isValid={[Function]}
          isValidValue={[Function]}
          mapping={null}
          onChange={[Function]}
          onError={[Function]}
          onInvalid={[Function]}
          onInvalidSubmit={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
          onValid={[Function]}
          onValidSubmit={[Function]}
          preventExternalInvalidation={false}
          resetValue={[Function]}
          setValidations={[Function]}
          setValue={[Function]}
          showError={[Function]}
          showRequired={[Function]}
          validationErrors={null}
        >
          <Formsy(Input)
            innerRef={[Function]}
            name="email"
            placeholder="Email"
            required={false}
            typeOfInput="email"
            validationError="This is not a valid email"
            validationErrors={
              Object {
                "isEmail": "This is not a valid email",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "input must be longer than 5 character",
              }
            }
            validations={
              Object {
                "isEmail": true,
                "maxLength": 48,
                "minLength": 5,
              }
            }
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="password"
            placeholder="Password"
            required={true}
            typeOfInput="password"
            validationError=""
            validationErrors={
              Object {
                "hasLowerCase": "Must have lowercase letter or you can
                     use a passphrase of minimum length 16 characters",
                "hasNumber": "Must have a number or you can
                     use a passphrase of minimum length 16 characters",
                "hasSpecialCharacter": "Must have a special character or 
                    you can use a passphrase of minimum length 16 characters",
                "hasUpperCase": "Must have uppercase letter or you can 
                    use a passphrase of minimum length 16 characters",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "Must have at least 8 characters",
              }
            }
            validations={
              Object {
                "hasLowerCase": [Function],
                "hasNumber": [Function],
                "hasSpecialCharacter": [Function],
                "hasUpperCase": [Function],
                "maxLength": 48,
                "minLength": 8,
              }
            }
          />
          <button
            className="btn login btn-disabled"
            disabled={true}
            type="submit"
          >
            Continue
          </button>
        </Formsy>
        <div
          className="stacked-text"
        >
          <Link
            replace={false}
            to="/signup"
          >
            <p
              id="signup"
            >
              Don't have an account? Click here to create one
            </p>
          </Link>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Login Component renders correctly 3`] = `
<div
  className="canvas"
>
  <div
    className="container2"
  >
    <header
      className="header"
    >
      <h2
        className="heading"
      >
        Book A Meal
      </h2>
    </header>
    <main>
      <div
        className="welcome"
      >
        <h4
          className="welcome-text"
        >
          Welcome!
        </h4>
        <h5>
          Sign in by entering the information below
        </h5>
      </div>
      <div
        className="form-box"
      >
        <Formsy
          className="form"
          disabled={false}
          getErrorMessage={[Function]}
          getErrorMessages={[Function]}
          getValue={[Function]}
          hasValue={[Function]}
          isFormDisabled={[Function]}
          isFormSubmitted={[Function]}
          isPristine={[Function]}
          isRequired={[Function]}
          isValid={[Function]}
          isValidValue={[Function]}
          mapping={null}
          onChange={[Function]}
          onError={[Function]}
          onInvalid={[Function]}
          onInvalidSubmit={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
          onValid={[Function]}
          onValidSubmit={[Function]}
          preventExternalInvalidation={false}
          resetValue={[Function]}
          setValidations={[Function]}
          setValue={[Function]}
          showError={[Function]}
          showRequired={[Function]}
          validationErrors={null}
        >
          <Formsy(Input)
            innerRef={[Function]}
            name="email"
            placeholder="Email"
            required={false}
            typeOfInput="email"
            validationError="This is not a valid email"
            validationErrors={
              Object {
                "isEmail": "This is not a valid email",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "input must be longer than 5 character",
              }
            }
            validations={
              Object {
                "isEmail": true,
                "maxLength": 48,
                "minLength": 5,
              }
            }
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="password"
            placeholder="Password"
            required={true}
            typeOfInput="password"
            validationError=""
            validationErrors={
              Object {
                "hasLowerCase": "Must have lowercase letter or you can
                     use a passphrase of minimum length 16 characters",
                "hasNumber": "Must have a number or you can
                     use a passphrase of minimum length 16 characters",
                "hasSpecialCharacter": "Must have a special character or 
                    you can use a passphrase of minimum length 16 characters",
                "hasUpperCase": "Must have uppercase letter or you can 
                    use a passphrase of minimum length 16 characters",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "Must have at least 8 characters",
              }
            }
            validations={
              Object {
                "hasLowerCase": [Function],
                "hasNumber": [Function],
                "hasSpecialCharacter": [Function],
                "hasUpperCase": [Function],
                "maxLength": 48,
                "minLength": 8,
              }
            }
          />
          <button
            className="btn login btn-disabled"
            disabled={true}
            type="submit"
          >
            Continue
          </button>
        </Formsy>
        <div
          className="stacked-text"
        >
          <Link
            replace={false}
            to="/signup"
          >
            <p
              id="signup"
            >
              Don't have an account? Click here to create one
            </p>
          </Link>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Login Component should map state to props 1`] = `
Object {
  "authenticated": false,
  "loginError": "",
}
`;
